test-harness/README.md

```diff
@@ -1,39 +1,75 @@
-## Generating binaries
+# Test Harness
 
-For anything to work, the binaries have to be generated first with: `yarn build.binaries`. This will generate three binaries.
+## Prerequisites
 
-### Alternatively, run any of the following:
+* Install the project dependencies with `yarn`
+* Generate the binary for your platform with `yarn build.binary`. This will *also* compile the project from TS -> JS
 
-1. `yarn build.binary` will assume the host machine
-2. `HOST=host yarn build.binary` will generate a binary for the current host machine
-3. `HOST=node10-linux-x64 yarn build.binary` will generate a binary for node10-linux-x64
+## Running the suite
 
-Before running any of the above, please make sure that the dependencies (from `package.json`) are installed.
-You can do so by running `yarn` or `npm install`.
-If the dependencies were not installed, you might experience `Error: command mock not found` issue.
+Run `yarn test.harness` from your terminal
 
-`HOST` variable is used to specify what binary we want to build.
-`HOST` might be set to anything that is considered a target in `pkg` [supports](https://github.com/zeit/pkg#targets).
+## Adding a new test
 
-## To run tests, run any of the following:
+* Create a new file in the `./spec` directory. It can have _any_ name and _any_ extension, it does not really matter.
+* Use the following template and put your stuff:
 
-Start Prism with the target file you'd like and then run the command
+```
+====test====
+Test text, can be multi line.
+====spec====
+openapi 2/3 document
+====server====
+command line arguments to run Prism with
+====command====
+curl command to get the response
+====expect====
+expected output from curl
+```
 
-1. `yarn test.binary`
-2. `SPEC=./examples/petstore.oas2.json yarn test.binary`
-3. `SPEC=./examples/petstore.oas2.json PRISM_PORT=4011 yarn test.binary`
+### A real example?
 
-> `SPEC` can take comma delimited paths to specs
+```
+====test====
+Hello
+====spec====
+openapi: 3.0.2
+paths:
+  /todos:
+    get:
+      responses:
+        200:
+          description: Get Todo Items
+          content:
+            'application/json':
+              example: hello
+====server====
+mock -p 4010
+====command====
+curl -i -X GET http://localhost:4010/todos -H "accept: application/json"
+====expect====
+HTTP/1.1 200 OK
+content-type: application/json; charset=utf-8
+content-length: 7
+Date: Fri, 21 Jun 2019 09:25:34 GMT
```

Comment:
- Should be able to blacklist certain headers, why test Date?

Reply:
Because we're testing e2e - the response should have a date header.

Reply:
Fine, ignoring the Date header.

Do I have to calculate `content-length` then? If I am creating a test in the harness, the will need to be `content-length` calculated correctly when generating a Raw HTTP Response(that is all the output is from curl). Even if I start out with output from curl, I will need to edit it to fix the errors, which in a lot of cases will require a `content-length` change.

Reply:
Yep certain headers will need to be removed for comparison by default, but its fine to have them in the response here. We want minimal mucking about for users/support/business to create these scenarios. It should be copy and pasteable by anyone without technical understanding.

Reply:
I would definitely not expect anyone to be writing these tests by hands no! That would be awful. These are gold master files. https://codeclimate.com/blog/gold-master-testing/

You run it, you store the response, then if it changes you get a diff.

This is not a test suite in the classic sense, we are not writing assertions.

Reply:
Just to be clear, I am not advocating that this be a suite in a classic sense.

Reply:
Gavel will ignore most of the headers and just evaluate the one that actually make sense (such as content type and some other right now I do not remember)